File: frontend/src/lib/tauri-api/merge-readiness.ts
import type { WorkspaceMergeReadiness } from '../../types/merge-readiness';
import { invokeCommand } from './client';

export function getWorkspaceMergeReadiness(workspaceId: string): Promise<WorkspaceMergeReadiness> {
  return invokeCommand<WorkspaceMergeReadiness>('get_workspace_merge_readiness', { workspaceId });
}

export function refreshWorkspaceMergeReadiness(workspaceId: string): Promise<WorkspaceMergeReadiness> {
  return invokeCommand<WorkspaceMergeReadiness>('refresh_workspace_merge_readiness', { workspaceId });
}

File: src-tauri/src/commands/merge_readiness.rs
use tauri::State;

use crate::models::WorkspaceMergeReadiness;
use crate::services::merge_readiness_service;
use crate::state::AppState;

#[tauri::command]
pub fn get_workspace_merge_readiness(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceMergeReadiness, String> {
    merge_readiness_service::get_workspace_merge_readiness(&state, &workspace_id)
}

#[tauri::command]
pub fn refresh_workspace_merge_readiness(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceMergeReadiness, String> {
    merge_readiness_service::refresh_workspace_merge_readiness(&state, &workspace_id)
}
